### PR TITLE
std: use relative imports for stdlib

### PR DESCRIPTION
**Pull request by tauoverpi**
This changes all imports of `std` to be relative rather than import it as an external module.

### CODE REVIEW

**Comment by thejoshwolfe** [comment]
What's the motivation for this?

**Comment by tauoverpi** [comment]
Consistency as currently the standard library mixes import arbitrarily and it doesn't make much sense for a library to import itself when the root source file is available. The standard library also serves as an example of how to structure a multi-file project so cleaning up minor things like this should lead to less confusion for new users.

I did make sure to ask on [ziggit](https://ziggit.dev/t/consistent-import-of-the-std-module/1509) and on discord (while I had the account) regarding the import style where both resulted in the same answer of relative paths being the preferred.

**Review by andrewrk — CHANGES_REQUESTED** [review]
Please do not submit a pull request that contains lines of code you have not personally inspected for correctness.

File: lib/std/os.zig [hunk 1]
```diff
@@ -40,7 +40,7 @@ pub const wasi = @import("os/wasi.zig");
 pub const windows = @import("os/windows.zig");
 
 comptime {
-    assert(@import("std") == std); // std lib tests require --zig-lib-dir
+    assert(@import("std.zig") == std); // std lib tests require --zig-lib-dir
```
[the remark this answers is not in the document]
Comment (tauoverpi) [reply lib/std/os.zig]: Sorry, I missed this while going over the changes.

**Comment by thejoshwolfe** [comment]
Consistency isn't an end-goal, but it can be a means to an end. Inconsistency can also be a means to an end. One example is that the std lib tests do not fully work on my system, but the tests in the json subdirectory do. I intentionally chose which imports would go through `@import("std")` vs which would be relative so that i could test the json tests independently of the rest of the std lib tests with `zig test lib/std/json.zig`. There may have been other ways to accomplish that, but intentional "inconsistency" was the tool i chose to use at the time.

If the end-goal of consistency is to avoid confusion for new users, then it's worth discussing if this PR is the best approach. Perhaps a tutorial would be even more effective than this PR.

**Comment by squeek502** [comment]
> I intentionally chose which imports would go through `@import("std")` vs which would be relative so that i could test the json tests independently of the rest of the std lib tests with `zig test lib/std/json.zig`. There may have been other ways to accomplish that, but intentional "inconsistency" was the tool i chose to use at the time.

This is https://github.com/ziglang/zig/issues/14504 (and I agree that it's a very important use-case, but it seems like the solution will be an improved `--test-filter` rather than making `zig test lib/std/json.zig` work again, see https://github.com/ziglang/zig/pull/16831)

**Comment by thejoshwolfe** [comment]
After reading the existing discussions linked above, it sounds like my intentional inconsistency approach is not favored by andrew. I guess i'll have to find a new workaround for developing in that subdirectory.

**Comment by andrewrk** [comment]
I'm not against `@import("std")` if it solves a problem. I was just answering the question of which convention to favor.

**Comment by andrewrk** [comment]
I'm not super thrilled with big "refactoring" patches like these. They're really easy for core team members to make when the time comes, but they tend to have bugs when sent in by people looking to make easy contributions. They just make me spend a bunch of time reviewing spelling fixes and unimportant stuff when I could just do it myself faster, if that's what I decide to spend my time on.

And what reward do we get for merging it? Nothing. It's not even clear this is better than status quo.

I appreciate the intentions here, @tauoverpi. But it's not time to remove specks of dust off the floor, when we still need to do the dishes and clean the kitchen.

**Comment by squeek502** [comment]
Out of pure curiosity, I tried comparing the time taken to build the standard library tests with (1) all `@import("std.zig")` converted to `@import("std")`, and (2) all `@import("std")` converted to `@import("std.zig")`.

There was no difference:

```
Benchmark 1 (3 runs): zig test lib/std/std.zig --zig-lib-dir lib --main-pkg-path lib/std --test-no-exec
  measurement          mean ± σ            min … max           outliers         delta
  wall_time          34.6s  ± 1.40s     33.8s  … 36.2s           0 ( 0%)        0%
  peak_rss           2.79GB ± 62.3MB    2.75GB … 2.86GB          0 ( 0%)        0%
  cpu_cycles          143G  ± 7.11G      139G  …  151G           0 ( 0%)        0%
  instructions        217G  ± 9.93G      211G  …  228G           0 ( 0%)        0%
  cache_references   15.1G  ±  597M     14.7G  … 15.8G           0 ( 0%)        0%
  cache_misses       3.04G  ±  138M     2.94G  … 3.19G           0 ( 0%)        0%
  branch_misses       850M  ± 53.0M      816M  …  911M           0 ( 0%)        0%
Benchmark 2 (3 runs): zig test lib2/std/std.zig --zig-lib-dir lib2 --main-pkg-path lib2/std --test-no-exec
  measurement          mean ± σ            min … max           outliers         delta
  wall_time          34.8s  ± 1.20s     34.1s  … 36.1s           0 ( 0%)          +  0.4% ±  8.5%
  peak_rss           2.79GB ± 73.1MB    2.75GB … 2.88GB          0 ( 0%)          +  0.2% ±  5.5%
  cpu_cycles          143G  ± 6.55G      139G  …  151G           0 ( 0%)          -  0.1% ± 10.8%
  instructions        218G  ± 11.0G      211G  …  230G           0 ( 0%)          +  0.3% ± 11.0%
  cache_references   15.1G  ±  536M     14.8G  … 15.7G           0 ( 0%)          -  0.3% ±  8.5%
  cache_misses       3.04G  ±  127M     2.96G  … 3.19G           0 ( 0%)          +  0.1% ±  9.9%
  branch_misses       841M  ± 48.8M      812M  …  897M           0 ( 0%)          -  1.0% ± 13.6%
```

**Pull request closed without merging**